src/logic/calculate.js

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable no-param-reassign */
+import operationsMethod from './operate';
+
+const calculate = (data, name) => {
+  const digits = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+  const operations = ['+', '-', '/', '*'];
+  const { operate } = operationsMethod;
+  if (digits.includes(name) && data.next === null) {
+    data.next = name;
```

Comment:
**OPTIONAL**

- [ ] You can use object destructuring here to get `next`, `operation` and `total`.
```JS
const { next, operation, total } = data;
```